packages/nextjs/app/builders/page.tsx

```diff
@@ -0,0 +1,108 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import Link from "next/link";
+import { BUILDERS } from "./constants";
+import type { NextPage } from "next";
+import { Address } from "~~/components/scaffold-eth";
+import { useScaffoldEventHistory } from "~~/hooks/scaffold-eth";
+
+type Builder = {
+  name: string;
+  address: string;
+  hasPersonalPage: boolean;
+};
+
+const Builders: NextPage = () => {
+  const [builders, setBuilders] = useState([] as Builder[]);
+
+  const { data: events } = useScaffoldEventHistory({
+    contractName: "BatchRegistry",
+    eventName: "CheckedIn",
+    fromBlock: 126461494n,
+  });
+
+  useEffect(() => {
+    const fetchBuilders = async () => {
+      if (!events) return;
+
+      // Filter out duplicate addresses
+      const filteredAddresses: string[] = [];
+      events.forEach(event => {
+        const address = event.args.builder;
+        if (!address) return;
+        if (!filteredAddresses.find(addr => address === addr)) filteredAddresses.push(address);
+      });
+
+      const builders = await Promise.all(
+        filteredAddresses.map(async address => {
+          // Check if the builder has a personal page
+          const resp = await fetch(`builders/${address}`);
```

Comment:
I am getting this error thrown into the console:
<img width="529" alt="Screenshot 2024-10-29 at 14 23 27" src="https://github.com/user-attachments/assets/a785459c-2f81-4a49-a891-7e667a8a3baa">

Reply:
@phipsae That `fetch` checks if a builder has a published profile so the 404 is expected behaviour. However, by now I think that check if obsolete since all the builder profiles have been merged. Should I remove it?

Reply:
Are there any other options you can think of? (Note: not all builders have profiles yet.)

Hint: You could consider making your `page.tsx` a server component by omitting the `"use client"`. This would allow you to access the fs module and read the created folders directly. Additionally, you could create a separate client component (with `"use client"`) for the members list, passing the data retrieved from the server component into it. And then you know which builders have already created a profile. 

Let me know if this makes sense. In general we never want to have console log errors.

Reply:
@pbkompasz there was just a little eel lint error with the dependencies in the useEffect array, but I fixed it.